demo: Convert JSON to protobuf.Value

miretskiy opened the pull request:
Demo conversion of arbitrary JSON to protobuf.Value Can easily be extended to support any crdb type.

```
 select crdb_internal.row_to_proto(rides)) from rides
limit 1;
                                                                                                                                                                                                                                                                                                                                                                                                crdb_internal.json_to_pbvalue
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  \x2a89030a230a0a73746172745f74696d6512151a13323031382d31322d31335430333a30343a30350a310a0d73746172745f6164647265737312201a1e31393035204368726973746f70686572204c6f636b73204170742e2037370a320a0872696465725f696412261a2463323866356332382d663563322d343030302d383030302d3030303030303030303032360a140a07726576656e756512091100000000004053400a2c0a02696412261a2461623032306334392d626135652d343830302d383030302d3030303030303030303134650a210a08656e645f74696d6512151a13323031382d31322d31345430383a30343a30350a340a0a76656869636c655f696412261a2461616161616161612d616161612d343830302d383030302d3030303030303030303030610a1b0a0c76656869636c655f63697479120b1a09616d7374657264616d0a2c0a0b656e645f61646472657373121d1a1b36363033372042656c696e646120506c617a61204170742e2039330a130a0463697479120b1a09616d7374657264616d
(1 row)

```

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/89955)
<!-- Reviewable:end -->

Commit:
sql: Add internal function to convert tuple to protocol message

Add a `crdb_internal.row_to_proto` function to convert
tuple to `google.protobuf.Struct` protocol message.

```
SELECT crdb_internal.row_to_proto(rides) FROM rides;

SELECT crdb_internal.row_to_proto(row(1, 'two', 'three'::bytea) FROM rides;
```

Epic: None
Release note: None

Commit:
wip

miretskiy commented:
Pushed rebased update for @dbist demo.

cockroach-teamcity commented:
[![CLA assistant check](https://cla.crdb.dev/pull/badge/not_signed)](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=89955) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=89955) before we can accept your contribution.<br/><hr/>**Yevgeniy Miretskiy** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.crdb.dev/check/cockroachdb/cockroach?pullRequest=89955) it.</sub>
